src/multihash.rs

```diff
@@ -285,6 +332,37 @@ where
     Ok((code, size as u8, digest))
 }
 
+/// Reads a multihash from a byte slice that contains a full multihash (code, size and the digest)
+///
+/// Returns the code, size and the digest. The size is the actual size and not the
+/// maximum/allocated size of the digest.
+///
+/// Currently the maximum size for a digest is 255 bytes.
+#[cfg(not(feature = "std"))]
+pub fn read_multihash<S>(r: &mut &[u8]) -> Result<(u64, u8, GenericArray<u8, S>), Error>
+where
+    S: Size,
+{
+    use unsigned_varint::decode;
+
+    let buf = *r;
+
+    let (code, buf) = decode::u64(buf)?;
+    let (size, buf) = decode::u64(buf)?;
+
+    if size > S::to_u64() || size > u8::MAX as u64 || size > buf.len() as u64 {
+        return Err(Error::InvalidSize(size));
+    }
+
+    *r = buf;
```

Comment:
I guess this reassignment is again related to the _double pointer_ argument? I find this to be an unintuitive side effect. Can this be made more explicit?